### PR TITLE
*: add erc20 event table

### PR DESCRIPTION
**Pull request by markya0616**

### CODE REVIEW

File: model/types.go [hunk 1]
```diff
@@ -161,3 +161,23 @@ func (s ERC20Storage) TableName() string {
...
+}
+
+// TableName retruns the table name of this erc20 contract
+func (s ERC20Transfer) TableName() string {
+	return ERC20TransferTableName(s.Address)
+}
+
+// ERC20TransferTableName returns its contract table
+func ERC20TransferTableName(address []byte) string {
+	return "erc20_transfer_" + hexutil.Encode(address)
```
Comment (kimmylin) [inline model/types.go]: add first 3 char of the erc20 contract name?

File: model/types.go [hunk 2]
```diff
@@ -161,3 +161,23 @@ func (s ERC20Storage) TableName() string {
...
+	Address     []byte `gorm:"-"`
+	BlockNumber int64  `gorm:"size:8;index"`
+	TxHash      []byte `gorm:"size:32;index"`
+	From        []byte `gorm:"size:20;index"`
+	To          []byte `gorm:"size:20;index"`
+	Value       string `gorm:"size:32"`
+}
+
+// TableName retruns the table name of this erc20 contract
+func (s ERC20Transfer) TableName() string {
```
Comment (tailingchen) [inline model/types.go]: How about introduce `ContractName`?

File: store/account/account.go [hunk 3]
```diff
@@ -58,9 +62,19 @@ func (t *store) InsertERC20(code *model.ERC20) error {
...
+		return err
+	}
+	// Create erc20 transfer event table
+	if err := t.db.CreateTable(model.ERC20Transfer{
 		Address: code.Address,
-	}).Error
+	}).Error; err != nil {
+		return err
+	}
+	// Create a table for this contract
```
Comment (kimmylin) [inline store/account/account.go]: rm redundant comment

File: store/account/account_test.go [hunk 4]
```diff
@@ -328,6 +331,56 @@ var _ = Describe("Account Database Test", func() {
 			}
 		})
 	})
+
+	Context("InsertERC20Transfer() & DeleteERC20Transfer()", func() {
+		It("deletes the right storage", func() {
```
Comment (kimmylin) [inline store/account/account_test.go]: deletes the right transfers

File: store/account/account.go [hunk 5]
```diff
@@ -36,6 +36,10 @@ type Store interface {
 	DeleteERC20Storage(address common.Address, from, to int64) error
 	LastSyncERC20Storage(address common.Address, blockNr int64) (result int64, err error)
 
+	// ERC 20 transfer events
+	InsertERC20Transfer(event *model.ERC20Transfer) error
+	DeleteERC20Transfer(address common.Address, from, to int64) error
```
Comment (kimmylin) [inline store/account/account.go]: how about FindERC20Transfer?

File: store/event_erc20.go [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+	"github.com/getamis/eth-indexer/common"
+	"github.com/getamis/eth-indexer/contracts"
+	"github.com/getamis/eth-indexer/model"
+)
+
+var (
+	erc20ABI, _ = abi.JSON(strings.NewReader(contracts.ERC20TokenABI))
+	// The sha3 signature of transfer event in erc20
+	// event Transfer(address indexed _from, address indexed _to, uint256 _value)
+	sh3TransferEvent = common.HexToBytes("0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef")
```
Comment (kimmylin) [inline store/event_erc20.go]: /s/sh3TransferEvent/sha3TransferEvent/

File: model/types.go (hunk 1 above)
Comment (markya0616) [reply model/types.go]: OK.

File: model/types.go (hunk 2 above)
Comment (markya0616) [reply model/types.go]: OK. I will add the first 3 char of contract name

File: store/account/account.go (hunk 3 above)
Comment (markya0616) [reply store/account/account.go]: OK

File: store/account/account_test.go (hunk 4 above)
Comment (markya0616) [reply store/account/account_test.go]: OK

File: store/account/account.go (hunk 5 above)
Comment (markya0616) [reply store/account/account.go]: We can add it in later PR if we need it.

File: store/event_erc20.go (hunk 6 above)
Comment (markya0616) [reply store/event_erc20.go]: OK

File: model/types.go (hunk 2 above)
Comment (markya0616) [reply model/types.go]: It will make our code more complex while query or insert data.

File: model/types.go (hunk 1 above)
Comment (markya0616) [reply model/types.go]: It will make our code more complex while query or insert data.

**Pull request merged**